File: 2598-smallest-missing-non-negative-integer-after-operations/2598-smallest-missing-non-negative-integer-after-operations.cpp
class Solution {
public:
    int findSmallestInteger(vector<int>& nums, int value) {
        map<int,int>mp;
        for (int& x : nums){
            int first = abs(x) % value;
            if (x < 0 && first != 0) {
                first = value - first;
            }
            mp[first]++;
        }
        int cnt = 0;
        int i = 0;
        while (mp[i] > 0) {
            mp[i++]--;
            if (i >= value) {
                i-=value;
                cnt++;
            }
        }
        return value * cnt + i;
    }
};

File: 2598-smallest-missing-non-negative-integer-after-operations/2598-smallest-missing-non-negative-integer-after-operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        mp = [0] * value
        for x in nums:
            mp[x % value] += 1
        ans = 0
        while mp[ans % value] > 0:
            mp[ans % value] -= 1
            ans += 1
        return ans
